tests/integration/cocoon/test_seed_sample_data.py

```diff
@@ -0,0 +1,276 @@
+            sub_holding_keys=[f"Transaction/{cls.scope}/strategy"],
+            file_type="csv",
+        )
+
+    def test_return_dict(self):
+
+        seed_data = self.seed_data
+
+        self.assertEqual(type(seed_data), dict)
+        self.assertEqual(set(seed_data.keys()), set(self.domain_list))
+
+
+class CocoonTestSeedDataWithOverrideCSV(unittest.TestCase, CocoonSeedDataTestsBase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        cls.scope = create_scope_id().replace("-", "_")
+        secrets_file = Path(__file__).parent.parent.parent.joinpath("secrets.json")
+        cls.api_factory = lusid.utilities.ApiClientFactory(
+            api_secrets_filename=secrets_file
+        )
+
+        cls.sample_data = pd.read_csv(
+            Path(__file__).parent.joinpath(
+                "data/seed_sample_data/sample_data_override.csv"
+            )
+        )
+
+        seed_data(
+            cls.api_factory,
+            ["portfolios", "instruments", "transactions"],
+            cls.scope,
+            Path(__file__).parent.joinpath(
+                "data/seed_sample_data/sample_data_override.csv"
+            ),
+            mappings=dict(
+                load_json_file(
+                    Path(__file__).parent.parent.parent.joinpath(
+                        "integration",
+                        "cocoon",
+                        "data",
+                        "seed_sample_data",
+                        "seed_sample_data_override.json",
+                    )
+                )
+            ),
+            sub_holding_keys=[f"Transaction/{cls.scope}/strategy"],
+            file_type="csv",
+        )
+
+
+class CocoonTestSeedDataNoOverrideExcel(unittest.TestCase, CocoonSeedDataTestsBase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        cls.scope = create_scope_id().replace("-", "_")
+        secrets_file = Path(__file__).parent.parent.parent.joinpath("secrets.json")
+        cls.api_factory = lusid.utilities.ApiClientFactory(
+            api_secrets_filename=secrets_file
+        )
+
+        cls.sample_data = pd.read_excel(
+            Path(__file__).parent.joinpath(
+                "data/seed_sample_data/sample_data_excel.xlsx"
+            )
+        )
+
+        seed_data(
+            cls.api_factory,
+            ["portfolios", "instruments", "transactions"],
+            cls.scope,
+            Path(__file__).parent.joinpath(
+                "data/seed_sample_data/sample_data_excel.xlsx"
+            ),
+            sub_holding_keys=[f"Transaction/{cls.scope}/strategy"],
+            file_type="xlsx",
+        )
+
+
+class CocoonTestSeedDataUnsupportedFile(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        cls.scope = create_scope_id().replace("-", "_")
+        secrets_file = Path(__file__).parent.parent.parent.joinpath("secrets.json")
+        cls.api_factory = lusid.utilities.ApiClientFactory(
```

Comment:
is this just so you can pass it through to `seed_data`? if so, its better to just mock it

Reply:
Sure, updated in latest PR.